common/navbar.js

```diff
@@ -284,6 +284,59 @@
 
                     scope.onLinkClick = onLinkClick(ConfigUtils, logService, UriUtils, $window);
 
+                    scope.showRidSearch = function () {
+                        return chaiseConfig.resolverImplicitCatalog !== null && chaiseConfig.hideSearchRID !== true
+                    }
+
+                    // RID search turned off in the cases of:
+                    //  - resolverImplicitCatalog == null
+                    //  - OR hideSearchRID == true
+                    //
+                    // The following cases need to be handled for the resolverImplicitCatalog value:
+                    //  - if catalogId is defined:
+                    //    - if resolverImplicitCatalog === null:        turn off config
+                    //    - if resolverImplicitCatalog === catalogId:   /id/RID
+                    //    - otherwise:                                  /id/catalogId/RID
+                    //  - else:
+                    //    - if resolverImplicitCatalog === null:        turn off config
+                    //    - if resolverImplicitCatalog is Not a Number: /id/RID
+                    //    - otherwise:                                  /id/resolverImplicitCatalog/RID
+                    scope.ridSearch = function () {
+                        var resolverId = chaiseConfig.resolverImplicitCatalog,
+                            url = "/id/", catId, splitId;
+
+                        if (isCatalogDefined(catalogId)) {
+                            splitId = UriUtils.splitVersionFromCatalog(catalogId);
+
+                            // use `/id/catalog/ridSearchTerm` format if:
+                            //   - resolver id is a different catalog id than current page
+                            //   - resolver id is NaN and !null
+                            if (isNaN(resolverId) || resolverId !== catalogId) {
+                                url += splitId.catalog + "/"
+                            }
+                        } else {
+                            // we don't have a catalog id, check if reolverId is a number and use as catalog id
+                            // basically catalogId !== resolverId for this case
+                            // use `/id/catalog/ridSearchTerm` format if:
+                            //   - resolver id is a number
+                            if (!isNaN(resolverId)) {
```

Comment:
This is not consistent with `resolvePermalink`. We're either going to add the current catalog or nothing. So you should remove this code path.